CONTRIBUTING.md

```diff
@@ -150,9 +150,46 @@ some_new_fancy_category         is correct
 - Filepaths will be used to dynamically create a directory of our algorithms.
 - Filepath validation will run on GitHub Actions to ensure compliance.
 
+##### Integrating CMake in a new directory
+
+In case a new directory is 100% required, `CMakeLists.txt` in the root directory needs to be updated, and a new `CMakeLists.txt` needs to be made within the new directory.\
+An example of how your new `CMakeLists.txt` file should look like. Note that if there are any extra libraries/setup required, you must include that in this file as well.
+
+```cmake
+# If necessary, use the RELATIVE flag, otherwise each source file may be listed
+# with full pathname. RELATIVE may makes it easier to extract an executable name
+# automatically.
+file( GLOB APP_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.c )
+# file( GLOB APP_SOURCES ${CMAKE_SOURCE_DIR}/*.c )
+# AUX_SOURCE_DIRECTORY(${CMAKE_CURRENT_SOURCE_DIR} APP_SOURCES)
+foreach( testsourcefile ${APP_SOURCES} )
+    # I used a simple string replace, to cut off .cpp.
+    string( REPLACE ".c" "" testname ${testsourcefile} )
+    add_executable( ${testname} ${testsourcefile} )
+
+    if(OpenMP_C_FOUND)
+        target_link_libraries(${testname} OpenMP::OpenMP_C)
+    endif()
+    if(MATH_LIBRARY)
+        target_link_libraries(${testname} ${MATH_LIBRARY})
+    endif()
+    install(TARGETS ${testname} DESTINATION "bin/<foldername>") # Folder name
+
+endforeach( testsourcefile ${APP_SOURCES} )
+```
+
+The `CMakeLists.txt` file in the root directory should be updated to include the new directory.\
+Include your new directory after the last subdirectory. Example:
+
+```cmake
+...
+add_subdirectory(divide_and_conquer)
+add_subdirectory(<foldername>)
+```
+
 #### Commit Guidelines
 
-- It is recommended to keep your changes grouped logically within individual commits. Maintainers find it easier to understand changes that are logically spilt across multiple commits. Try to modify just one or two files in the same directory.  Pull requests that span multiple directories are often rejected.
+- It is recommended to keep your changes grouped logically within individual commits. Maintainers find it easier to understand changes that are logically spilled across multiple commits. Try to modify just one or two files in the same directory. Pull requests that span multiple directories are often rejected.
```

Comment:
```suggestion
- It is recommended to keep your changes grouped logically within individual commits. Maintainers find it easier to understand changes that are logically split across multiple commits. Try to modify just one or two files in the same directory. Pull requests that span multiple directories are often rejected.
```

I think it was correct...

Reply:
Oh, I think `split` is American English, and `spilled` is British English. Which one should we use?

Reply:
Ah I think that's not quite right 😅 `split` means "breaking things apart" but `spilled` means (when talking about liquid) causing something to go over flow of container (something along those lines) 😄

Reply:
Hmm. Should I keep it as-is or change it back to `spilled`?

Reply:
Keep it as is 😄